[Pavitra] ip

pavz02 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

pavz02 committed:
Finished Level 1

pavz02 committed:
Finished level 2

pavz02 committed:
Finished Level 3

pavz02 committed:
Finished Level 4

pavz02 committed:
Finished Level 5

pavz02 committed:
Finished Level 6

pavz02 committed:
Cleaned up the code and added exceptions

pavz02 committed:
Updated input.txt and expected.txt to add test cases and the expected solution, Changed Duke to Luna

pavz02 committed:
trying to branch

pavz02 committed:
Finished Level 7 in a seperate branch

pavz02 committed:
Testing branching, nothing noteworthy has been changed.

pavz02 committed:
Revert "Testing branching, nothing noteworthy has been changed."

This reverts commit cf3a8b1db4def65d874c73e056444687215b2a41.

pavz02 committed:
Merge tag 'branch-Level-7'

pavz02 committed:
idk

pavz02 committed:
Merge branch 'branch-Level-8'

pavz02 committed:
Finished Level 8. Deadline and Event can now parse dates inputted in the format yyyy-mm-dd after the "/by" and "/at" strings respectively. Will add other date formats and times in later commits.

pavz02 committed:
Finished A-MoreOOP

pavz02 committed:
Added a dummy A-JUnit test for the class Duke. Will add JUnit tests for other classes soon.

pavz02 committed:
Tweaked the code to adhere to the Java coding standards.

pavz02 committed:
Finished A-JavaDoc. Added JavaDoc comments to all classes, constructors, and public methods.

pavz02 committed:
Finished Level 9. Added another keyword 'find' to the program.

pavz02 committed:
A-CodingStandard. Did it before the branch accidentally, but made the branch again. Only a few changes were made in this commit.

pavz02 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/TaskList.java
#	src/main/java/Ui.java

pavz02 committed:
Merge branch 'branch-A-CodingStandard'

pavz02 committed:
Added JavaDoc comments for the keyword 'Find' and the methods added to process it.

pavz02 committed:
Caught IO exceptions, inceased the date and time functionality for the Event and Deadline classes, made some general tweakings across all classes.

pavz02 committed:
Added some JUnit tests for the Ui class.

pavz02 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

pavz02 committed:
Added checkstyle in gradle and checkstyle-idea. Fixed all pre-existing checkstyle errors too.

pavz02 committed:
Added a GUI using JavaFx.

VisnuRavi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,99 @@
+     * Determines the keyword and processes the input based on the keyword.
+     */
+    static String parse(String inp) {
+        String[] spl = inp.split(" ", 2);
```
Could perhaps use a actual word to name the variable spl, as it is being used throughout the parse code

VisnuRavi replied:
also for the checkSplLength below

pavz02 replied:
Thank you for your review :)

VisnuRavi commented on src/test/java/UiTest.java:
```diff
@@ -0,0 +1,26 @@
+    private static final String LINE = "\n     <<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>><<>>\n";
+
+    @Test
+    public void CorrectTaskFormat_TypesOfTasks_FormatOfTask() {
```
I believe the starting letter should be lower caps according to coding standard

VisnuRavi reviewed (commented):
I think your code is great! I have mentioned just a few things that I would change, but other than that should be good to merge!

pavz02 committed:
Generated a new jar file.

pavz02 committed:
Added assertions to the code.

pavz02 committed:
Tried to improve the code quality.

pavz02 committed:
Merge pull request #1 from pavz02/branch-A-CodeQuality

?

pavz02 committed:
Merge branch 'master' into branch-A-Assertions

pavz02 committed:
Merge pull request #2 from pavz02/branch-A-Assertions

??

pavz02 committed:
Added the extension C-tagging. Can add tags via the command "tag *index of task* *description of tag*" or directly with the label "/tag" while entering the task. Multiple tags and deleting tags are not supported yet.

pavz02 committed:
Tweaked the gui by changing fonts and colours, making the profile picture circular, and highlighting errors.

pavz02 committed:
Added a product screenshot (Ui.png) to the docs folder.

pavz02 committed:
.

pavz02 committed:
Tried to enhance the GUI features, and also updated the user guide.

pavz02 committed:
..

pavz02 committed:
...

pavz02 committed:
Final commit

pavz02 committed:
Tag throws an error if no number is entered for the index number.
All number inputs are now checked if they are greater than 0.
User guide updated with the correct description of the tag command.
FXML files introduced with minor changes to the GUI.

pavz02 committed:
Updated some javadocs, command bye now correctly exits the program, greeting to the user on startup added, hardisk functionality added.

pavz02 committed:
Refactored Duke to Luna, fixed gradle, modified junit tests, fixed the parsing of the tag command,

pavz02 committed:
Very minor changes.
